src/sync/syncer2.ts

```diff
@@ -164,7 +166,7 @@ export class OnePubOneWorkspaceSyncer {
         this.state.isPushStreaming = false;
         this._bump();
     }
-    async startPullStream() {
+    startPullStream() {
```

Comment:
This function doesn't need to be async.  It contains an async function within it, as a callback to `onmessage`, but that function doesn't run while the main function is running.

E.g. this is ok
```ts
myNonAsyncFunction() {
    pullStream.onmessage(async (data) => { ... });
}
```
because the async function runs later, when a message arrives.

Reply:
The newly added `await` on line 190, in the `onmessage` handler, should be safe.

Adding an `await` into a function makes a point where other functions can run in the meantime while that one is waiting, so you have to make sure nothing is left in a dangerous / incomplete state while waiting.  In this case the only things that happen after the `await` are logs so there's no harm to be done if they end up running later after the world has changed around them.